aiohttp/http_parser.py

```diff
@@ -29,7 +29,15 @@
     'RawRequestMessage', 'RawResponseMessage')
 
 ASCIISET = set(string.printable)
-METHRE = re.compile('[A-Z0-9$-_.]+')
+
+# See https://tools.ietf.org/html/rfc7230#section-3.1.1
+# and https://tools.ietf.org/html/rfc7230#appendix-B
+#
+#     method = token
+#     tchar = "!" / "#" / "$" / "%" / "&" / "'" / "*" / "+" / "-" / "." /
+#             "^" / "_" / "`" / "|" / "~" / DIGIT / ALPHA
+#     token = 1*tchar
+METHRE = re.compile("[!#$%&'*+\-.^_`|~0-9A-Za-z]+")
```

Comment:
I'd use a raw-string literal `r''`

Reply:
Also, `+` and `*` might also need escaping and `|` needs escaping (otherwise it divides groups in `[]`)

Reply:
Oh, and `.` might stand for any single character.

Reply:
Probably they have restricted meanings within `[]`, but IMHO better safe than sorry.

Reply:
All true, but not for `[]` sets.
```
>>> import re
>>> re.match('[.]', 't')
>>> re.match('[.]', 't') is None
True
```

Reply:
digits can be replaced with `\d` (to match style below)